src/optimalcontrol/CMakeLists.txt

```diff
@@ -72,6 +96,9 @@ install(TARGETS ${libraryname}
         PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iDynTree)
 
 install(FILES ${INTEGRATORS_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iDynTree/Integrators)
+install(FILES ${OPTIMIZERS_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iDynTree/Optimizers)
+install(FILES ${OCSOLVERS_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iDynTree/OCSolvers)
+
 
 set_property(GLOBAL APPEND PROPERTY ${VARS_PREFIX}_TARGETS ${libraryname})
 set_property(GLOBAL APPEND PROPERTY IDYNTREE_TREE_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include)
```

Comment:
Random question: does this include folder target the BUILD or INSTALL interface? Why is this needed in addition to `target_include_directories`?

Reply:
This is to install files, not to link them to the target. The problem is that I wanted to keep the same tree structure also when installing. Using the default install interface, it was keeping everything in the same folder, loosing the tree structure I have in ``src``.